### ISSUE TITLE
Reference to location within data in SchemaError

### ISSUE THREAD

**Issue by demux** [keleshev/schema#93]
I would like SchemaError to tell me where in the data structure the error occurred.

Something like:

    # Schema:
    Schema({
        'key1': {
            'child1': {
                'child2': [
                    {'pedo': bool}
                ]
            }
        },
        Optional('key2'): {
            'foo': str
        }
    })

    # Data:
    {
        'key1': {
            'child1': {
                'child2': [
                    {'pedo': 1},
                    {'pedo': false},
                ]
            }
        }
    }

    # Error message:
    {
        'key1': {
            'child1': {
                'child2': {
                    0: {'pedo': 'this value should be an instance of bool not int'}
                }
            }
        }
    }

This way I would be able to find the location of the error in my parsed YAML file. It's also just a lot more helpful error message. Something like `1 should be bool` tells me almost nothing.

**Comment by danielmaxx** [comment]
I agree. The current message is quite useless. Having a simple list with the path to the error would be super helpful.

**Comment by vlcinsky** [comment]
Would be really helpful.

**Comment by timtan** [comment]
+1

**Comment by kurtbrose** [comment]
Totally agree, I'm going to take a shot at a PR.

**Comment by kurtbrose** [comment]
looks like this may be handled recently by this: https://github.com/keleshev/schema/commit/61cc0bf84dd3efc830a366fb200c386bae35c243
